core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitorActor.scala

```diff
@@ -277,12 +278,26 @@ trait SubmissionMonitor extends FutureSupport with LazyLogging with RawlsInstrum
   private def execServiceOutputs(workflowRec: WorkflowRecord, petUser: UserInfo)(implicit executionContext: ExecutionContext): Future[Option[(WorkflowRecord, Option[ExecutionServiceOutputs])]] = {
     WorkflowStatuses.withName(workflowRec.status) match {
       case WorkflowStatuses.Succeeded =>
+        // asynchronously upload metadata to GCS.
+        // Currently, this is only used for hamm, and may potentially be removed from RAWLS if design changes
+        workflowRec.externalId.traverse{
+          workflowId =>
+            uploadMetadataToGCS(workflowRec.submissionId.toString, workflowId, petUser)
+        }
         executionServiceCluster.outputs(workflowRec, petUser).map(outputs => Option((workflowRec, Option(outputs))))
-
       case _ => Future.successful(Option((workflowRec, None)))
     }
   }
 
+  private def uploadMetadataToGCS(submissionId: String, workflowId: String, petUser: UserInfo)(implicit executionContext: ExecutionContext): Future[Unit] = {
+    for{
+      metadata <- executionServiceCluster.callLevelMetadataForCostCalculation(submissionId, workflowId, None, petUser)
+//    If this is a subworkflow, we don't upload its metadata; else we upload it to GCS
+      _ <- if(metadata.fields.keys.toList.contains("parentWorkflowId")) Future.successful(())
```

Comment:
I don't think this will ever be true, a) we don't ask for the parentWorkflowId key and b) we only monitor the top level workflow

Reply:
I don't follow...Do you mean https://github.com/broadinstitute/rawls/pull/1038/files#diff-90caa3307e9614c29824ffa11e2fef55R285 already only has top level workflow?

Reply:
I mean the condition `metadata.fields.keys.toList.contains("parentWorkflowId")` will never evaluate to true

Reply:
the only workflows that rawls monitors are the top level workflows, I don't see how we would hit this code path with a sub workflow

But besides that, in `callLevelMetadataForCostCalculation` `parentWorkflowId` is not one of the keys requested. So even it were a sub workflow would `parentWorkflowId` exist?